x-pack/plugins/security_solution/public/common/lib/kibana/hooks.ts

```diff
@@ -51,57 +51,66 @@ export interface AuthenticatedElasticUser {
 }
 
 export const useCurrentUser = (): AuthenticatedElasticUser | null => {
+  const isMounted = useRef(false);
```

Comment:
Why did we have the need to change this hook?

Reply:
This is to avoid `Can't perform React state update on an unmounted component`. As there is an async api call the parent component can be unmounted when the async api call is done so we need to ensure the component is mounted to avoid this cain of warnings.

Reply:
ahh. Ok. so it has nothing to do with your changes, you are just fixing that error issue we see in the browser from time to time. 👍